spock-core/src/main/java/org/spockframework/runtime/DataIteratorFactory.java

```diff
@@ -93,6 +93,29 @@ protected int estimateNumIterations(Object[] dataProviders) {
       return result == Integer.MAX_VALUE ? UNKNOWN_ITERATIONS : result;
     }
 
+    protected int estimateNumIterations(IDataIterator[] dataProviderIterators) {
```

Comment:
I would guess it should have a `@Nullable` anotation, when it can be null in line 101.